### PR TITLE
Add shopping cart model

### PR DESCRIPTION
**Pull request by ArtemMakovskyy**

### CODE REVIEW

File: src/main/java/online/book/store/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.stereotype.Component;
+
+@SpringBootApplication
+@RequiredArgsConstructor
+public class OnlineBookStoreApplication {
+    private static final Logger logger = LogManager.getLogger(OnlineBookStoreApplication.class);
+
+    //    private final BookRepository bookRepository;
+    //    private final UserRepository userRepository;
+    //    private final ShoppingCartRepository shoppingCartRepository;
```
Comment (sokimaaa) [inline src/main/java/online/book/store/OnlineBookStoreApplication.java]: remove unused comments

File: src/main/java/online/book/store/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Category category = (Category) o;
+        return Objects.equals(id, category.id)
+                && Objects.equals(name, category.name)
+                && Objects.equals(description, category.description);
```
Comment (sokimaaa) [inline src/main/java/online/book/store/model/Category.java]:
```suggestion
return Objects.equals(id, category.id)
                && Objects.equals(name, category.name);
```

File: src/main/java/online/book/store/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return shoppingCartService.getShoppingCartDtoByUserId(user.getId());
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(summary = "Add book to the shopping cart.",
+            description = "Add book with quantity to the shopping cart")
+    public AddBookToTheShoppingCartDto addBookToTheShoppingCart(
+            Authentication authentication,
+            @RequestBody AddBookToTheShoppingCartDto addBookDto) {
```
Comment (sokimaaa) [inline src/main/java/online/book/store/controller/ShoppingCartController.java]: why do not validate addBookDto?

File: src/main/java/online/book/store/dto/cart/item/AddBookToTheShoppingCartDto.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package online.book.store.dto.cart.item;
+
+import lombok.Data;
+
+@Data
+public class AddBookToTheShoppingCartDto {
+    private Long bookId;
+    private int quantity;
```
Comment (sokimaaa) [inline src/main/java/online/book/store/dto/cart/item/AddBookToTheShoppingCartDto.java:12]: why validation annotations are not present here? e.g.
https://github.com/NotNull, @min

File: src/main/java/online/book/store/OnlineBookStoreApplication.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        logger.info("API Documentation Overview: http://localhost:8080/api/swagger-ui/index.html#/");
+    }
+
+    @Component
+    public class ApplicationRunnerImpl implements ApplicationRunner {
+        @Override
+        public void run(ApplicationArguments args) throws Exception {
+
+        }
+    }
```
Comment (sokimaaa) [inline src/main/java/online/book/store/OnlineBookStoreApplication.java]: this code is doing something? if not should be removed

File: src/main/java/online/book/store/controller/AuthenticationController.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @Operation(summary = "Registered user login.",
+            description = "Input email address and password to login.")
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
+        System.out.println(requestDto);
```
Comment (aaananas) [inline src/main/java/online/book/store/controller/AuthenticationController.java]: if you want to output something - use logs

in this case, it is not recommended, that you expose passwords in logs. It is sensitive information

File: src/main/java/online/book/store/mapper/BookMapper.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    @AfterMapping
+    default void setCategoriesIntoBook(@MappingTarget Book book, CreateBookRequestDto createDto) {
+        if (!createDto.getCategories().isEmpty()) {
+            final Set<Category> categories = new HashSet<>();
+            for (Long id : createDto.getCategories()) {
+                Category category = new Category(id);
+                categories.add(category);
+            }
+            book.setCategories(categories);
+        }
```
Comment (aaananas) [inline src/main/java/online/book/store/mapper/BookMapper.java]:
```suggestion
            final Set<Category> categories = new HashSet<>();
            for (Long id : createDto.getCategories()) {
                Category category = new Category(id);
                categories.add(category);
            }
            book.setCategories(categories);
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Pull request closed without merging**